iam/snippets/policy_set.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START iam_set_policy]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+// setPolicy sets IAM policy for a project.
+func setPolicy(w io.Writer, projectID string, policy *cloudresourcemanager.Policy) error {
```

Comment:
issue: construct a `Policy` object here so that devs can see how it's done.

Reply:
Why make it different from all other languages https://cloud.google.com/iam/docs/samples/iam-set-policy? From the fixit instruction, the goal is to add consistent samples.

Reply:
Let's try to improve the samples gradually over time. The sample here is to set a Policy, which implies that we should show how to create a Policy message.

Also see the Sample Guidelines:
https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments